torch/_inductor/kernel/flex_decoding.py

```diff
@@ -294,11 +297,13 @@ def flex_decoding_grid(batch_size, kv_heads, gqa_group_size, n_keys, d_model, me
 )
 
 
-def get_split_k(B: int, H: int, Mk: int, SM: int = 128) -> int:
-    """Heuristic for the number of splits from xformer"""
+def get_split_k(B: int, H: int, Mk: int) -> int:
+    num_SM = torch.cuda.get_device_properties("cuda").multi_processor_count
```

Comment:
when this lands we should update: https://github.com/pytorch/pytorch/pull/142101/files#diff-8dfc175f817a04eb10c183509564129a91f32db31190e7b4ad3fd9653337c6f6R1154
maybe just add a todo so somone can pick up if this lands first